README.md
# A basic auth scaffolding for laravel using Tailwindcss and Fortify

[![Latest Version on Packagist](https://img.shields.io/packagist/v/awcodes/laravel-auth-scaffold.svg?style=flat-square)](https://packagist.org/packages/awcodes/laravel-auth-scaffold)
[![GitHub Tests Action Status](https://img.shields.io/github/workflow/status/awcodes/laravel-auth-scaffold/run-tests?label=tests)](https://github.com/awcodes/laravel-auth-scaffold/actions?query=workflow%3Arun-tests+branch%3Amaster)
[![Total Downloads](https://img.shields.io/packagist/dt/awcodes/laravel-auth-scaffold.svg?style=flat-square)](https://packagist.org/packages/awcodes/laravel-auth-scaffold)


This is just a basic auth scaffolding for Laravel Fortify. If you want to use it great, if not, so be it.

## Installation

You can install the package via composer:

```bash
composer require awcodes/laravel-auth-scaffold
```

You can publish the views with:
```bash
php artisan vendor:publish --provider="Awcodes\Auth\AuthServiceProvider" --tag="views"
```

## Testing

``` bash
composer test
```

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information on what has changed recently.

## Contributing

Please see [CONTRIBUTING](.github/CONTRIBUTING.md) for details.

## Security Vulnerabilities

Please review [our security policy](../../security/policy) on how to report security vulnerabilities.

## Credits

- [awcodes](https://github.com/awcodes)
- [All Contributors](../../contributors)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `laravel-auth-scaffold` will be documented in this file

## 1.0.0 - 202X-XX-XX

- initial release


tests/TestCase.php
<?php

namespace Awcodes\Auth\Tests;

use Awcodes\Auth\AuthServiceProvider;
use Illuminate\Database\Eloquent\Factories\Factory;
use Orchestra\Testbench\TestCase as Orchestra;

class TestCase extends Orchestra
{
    public function setUp(): void
    {
        parent::setUp();

        Factory::guessFactoryNamesUsing(
            fn (string $modelName) => 'Awcodes\\Auth\\Database\\Factories\\'.class_basename($modelName).'Factory'
        );
    }

    protected function getPackageProviders($app)
    {
        return [
            AuthServiceProvider::class,
        ];
    }

    public function getEnvironmentSetUp($app)
    {
        $app['config']->set('database.default', 'sqlite');
        $app['config']->set('database.connections.sqlite', [
            'driver' => 'sqlite',
            'database' => ':memory:',
            'prefix' => '',
        ]);

        /*
        include_once __DIR__.'/../database/migrations/create_laravel_auth_scaffold_table.php.stub';
        (new \CreatePackageTable())->up();
        */
    }
}


tests/ExampleTest.php
<?php

namespace Awcodes\Auth\Tests;

class ExampleTest extends TestCase
{
    /** @test */
    public function true_is_true()
    {
        $this->assertTrue(true);
    }
}


src/AuthServiceProvider.php
<?php

namespace Awcodes\Auth;

use Illuminate\Support\ServiceProvider;
use Laravel\Fortify\Fortify;

class AuthServiceProvider extends ServiceProvider
{
    public function boot()
    {
        Fortify::createUsersUsing(CreateNewUser::class);
        Fortify::updateUserProfileInformationUsing(UpdateUserProfileInformation::class);
        Fortify::updateUserPasswordsUsing(UpdateUserPassword::class);
        Fortify::resetUserPasswordsUsing(ResetUserPassword::class);

        Fortify::loginView(function () {
            return view('auth::login');
        });

        Fortify::requestPasswordResetLinkView(function () {
            return view('auth::forgot-password');
        });

        Fortify::resetPasswordView(function ($request) {
            return view('auth::reset-password', ['request' => $request]);
        });

        if ($this->app->runningInConsole()) {
            $this->publishes([
                __DIR__ . '/../resources/views' => base_path('resources/views/vendor/auth'),
                __DIR__ . '/../resources/css' => public_path('vendor/auth/css'),
            ], 'awcodes-auth-views');
        }

        $this->loadViewsFrom(__DIR__ . '/../resources/views', 'auth');
    }

    public function register()
    {
    }
}


resources/views/layout.blade.php
<!DOCTYPE html>
<html lang="en" class="h-full">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>
        @isset($title)
        {{ $title }} |
        @endisset
        {{ config('app.name') }}
    </title>
    <link href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;500;700&display=swap" rel="stylesheet">
    <link href="{{ asset('vendor/auth/css/auth.css') }}" rel="stylesheet" />
</head>

<body class="bg-gray-900 text-white antialised h-full flex font-sans">
    <div class="w-full max-w-md h-full flex flex-col justify-center p-10 border-r-4 border-pink-600">
        <h1 class="text-3xl font-bold mb-6">{{ config('app.name') }}</h1>
        @yield('body')
    </div>
    <div class="relative flex-1 hidden w-0 lg:block">
        <img class="absolute inset-0 object-cover w-full h-full" src="https://source.unsplash.com/featured/?wallpaper" alt="">
    </div>
</body>

</html>

resources/views/reset-password.blade.php
@extends('auth::layout', [
'title' => 'Reset your password'
])

@section('body')
<h2 class="text-2xl">Reset your password</h2>
<p class="text-sm mt-2">Whoosah! Let's get you into your account.</p>
<form method="POST" action="/reset-password" class="mt-8">
    @csrf

    <input type="hidden" name="token" value="{{ $request->route('token') }}">

    <div class="space-y-6">
        <div>
            <label for="email" class="block text-sm font-medium leading-5 text-gray-300">
                Email address
            </label>
            <input id="email" type="email" name="email" value="{{ old('email', $request->email) }}" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded">
            @error('email')
            <div class="mb-8 text-sm text-red-300 mt-1" role="alert">
                {{ $message }}
            </div>
            @enderror
        </div>

        <div>
            <label for="password" class="block text-sm font-medium leading-5 text-gray-300">
                Password
            </label>
            <input id="password" type="password" name="password" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded">
            @error('password')
            <div class="mb-8 text-sm text-red-300 mt-1" role="alert">
                {{ $message }}
            </div>
            @enderror
        </div>

        <div>
            <label for="password-confirm" class="block text-sm font-medium leading-5 text-gray-300">
                Confirm Password
            </label>
            <input id="password-confirm" type="password" name="password_confirmation" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded">
        </div>
    </div>

    <div class="mt-12">
        <button type="submit" class="px-8 py-2 text-lg font-bold text-white bg-pink-600 rounded-full hover:bg-pink-500">
            Reset Password
        </button>
    </div>

    <div class="mt-8">
        <a href="/login" class="text-sm font-medium text-pink-500 hover:text-pink-400 hover:underline">
            Nevermind, I remember now.
        </a>
    </div>
</form>
@endsection

resources/views/forgot-password.blade.php
@extends('auth::layout', [
'title' => 'Forgot Password'
])

@section('body')
<h2 class="text-2xl">Forgot your password, huh?</h2>
<p class="text-sm mt-2">No worries! Just let me know your email address.</p>
<form method="POST" action="{{ route('password.email') }}" class="mt-8">
    @csrf

    <div>
        <label for="email" class="col-md-4 col-form-label text-md-right">Email</label>

        <div class="col-md-6">
            <input id="email" type="email" name="email" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded">
        </div>
    </div>

    <div class="mt-12">

        @error('email')
        <div class="mb-8 text-red-300" role="alert">
            <strong>Are you sure we know each other?</strong>
            <div class="text-sm">{{ $message }}</div>
        </div>
        @enderror

        @if (session('status'))
        <div class="mb-8 text-green-300" role="alert">
            {{ session('status') }}
        </div>
        @endif

        <button type="submit" class="px-8 py-2 text-lg font-bold text-white bg-pink-600 rounded-full hover:bg-pink-500">
            {{ __('Send Password Reset Link') }}
        </button>
    </div>

    <div class="mt-8">
        <a href="/login" class="text-sm font-medium text-pink-500 hover:text-pink-400 hover:underline">
            Nevermind, I remember now.
        </a>
    </div>
</form>
@endsection

resources/views/login.blade.php
@extends('auth::layout', [
'title' => 'Login'
])

@section('body')
<h2 class="text-2xl">Log in to your account</h2>
<form method="POST" action="/login" class="mt-8">
    @csrf
    <div class="space-y-6">
        <div>
            <label for="email" class="block text-sm font-medium leading-5 text-gray-300">
                Email
            </label>
            <input id="email" type="email" name="email" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded" value="{{ old('email') }}">
        </div>

        <div>
            <label for="password" class="block text-sm font-medium leading-5 text-gray-300">
                Password
            </label>
            <input id="password" type="password" name="password" required class="mt-1 block w-full p-3 text-gray-900 placeholder-gray-400 rounded">
        </div>
    </div>

    <div class="mt-12">
        @if($errors->any())
        <div class="mb-8 text-red-300" role="alert">
            <strong>Are you sure we know each other?</strong>
            <div class="text-sm">I couldn't log you in with the provided credentials.</div>
        </div>
        @endif

        @if (session('status'))
        <div class="mb-8 text-green-300" role="alert">
            {{ session('status') }}
        </div>
        @endif

        <button type="submit" class="px-8 py-2 text-lg font-bold text-white bg-pink-600 rounded-full hover:bg-pink-500">
            Sign in
        </button>
    </div>

    <div class="mt-8">
        <a href="/forgot-password" class="text-sm font-medium text-pink-500 hover:text-pink-400 hover:underline">
            I'm lame and don't know my password
        </a>
    </div>
</form>
@endsection